### PR TITLE
better handling of cache miss

### PR DESCRIPTION
**Pull request by mmaxim**
Patch does the following:

1.) Doesn't cancel the local thread if the remote thread fails.
2.) Modify the chat thread message to take a potentially null second parameter when we miss the local cache.

cc @chrisnojima

### CODE REVIEW

File: go/service/chat_local.go [hunk 1]
```diff
@@ -328,6 +329,9 @@ func (h *chatLocalHandler) GetThreadNonblock(ctx context.Context, arg chat1.GetT
 			SessionID: arg.SessionID,
 			Thread:    remoteThread,
 		})
+
+		// This means we transmitted with success, so cancel local thread
+		cancel()
```
Comment (patrickxb) [inline go/service/chat_local.go]: What's the difference calling `cancel` here?

Comment (mmaxim) [inline go/service/chat_local.go]: Used to do `defer cancel()` at the top, but I didn't want it to do that if we hit an error.

Comment (patrickxb) [inline go/service/chat_local.go]: I think it's a (potential) mem leak if cancel isn't called...

Comment (mmaxim) [inline go/service/chat_local.go]: Wouldn't everything get cleaned up when `bctx` gets collected?

Comment (mmaxim) [inline go/service/chat_local.go]: I could also just stick this at the end
```
select {
case <-bctx.Done():
default:
    cancel()
}
```

but I figured that would just get handled once the context died.

Comment (patrickxb) [inline go/service/chat_local.go]: I think it's required to call `cancel`:

> The WithCancel, WithDeadline, and WithTimeout functions take a Context (the parent) and return a derived Context (the child) and a CancelFunc. Calling the CancelFunc cancels the child and its children, removes the parent's reference to the child, and stops any associated timers. Failing to call the CancelFunc leaks the child and its children until the parent is canceled or the timer fires. The go vet tool checks that CancelFuncs are used on all control-flow paths.

https://golang.org/pkg/context/

Comment (mmaxim) [inline go/service/chat_local.go]: @patrickxb is right here, I was assuming too much about what goes on with these `CancelFunc` deals. Thanks!

**Comment by keybase-ci-visdiff** [comment]
The commits 0f8ce2b1a334...b9bf216f0c9f introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490045341622-0f8ce2b1a334-b9bf216f0c9f-linux/Markdown-emoji-0f8ce2b1a334.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490045341622-0f8ce2b1a334-b9bf216f0c9f-linux/Markdown-emoji-b9bf216f0c9f.png)  
   Markdown-emoji comparison
</summary>

**Pull request merged**